### PR TITLE
Use edition links for Worldwide Organisations

### PR DESCRIPTION
**Pull request by brucebolt**
In https://github.com/alphagov/whitehall/pull/8734 and https://github.com/alphagov/publishing-api/pull/2601, we have switched worldwide organisations to use edition links to fix a bug in editionable worldwide organisations.

This applies the changes needed to the frontend to reflect the lack of multi-level link expansion in edition links.

Whilst this change adds support for edition links, it also retains compatibility with non-edition links. This will allow us to deploy the frontend changes in advance of republishing the content items, preventing any downtime of these pages. The code for backward compatibility will be removed once the content items have all been republished with edition links (all the code to be removed has been marked with a comment so it is easy to identify).

[Trello card](https://trello.com/c/yNJwHw4K)

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

**Review by callumknights — APPROVED** [review]

File: app/presenters/worldwide_organisation_presenter.rb [hunk 1]
```diff
@@ -136,6 +147,18 @@ def sponsoring_organisations
 
 private
 
+  def contact_for_office(office_content_id)
+    contact_mapping = content_item.dig("details", "office_contact_associations").select { |office_contact_association|
+      office_contact_association["office_content_id"] == office_content_id
+    }.first
+
+    return unless contact_mapping
```
Comment (hannako) [inline app/presenters/worldwide_organisation_presenter.rb:155]: I notice that we don't perform this check in [`roles_for_person`](https://github.com/alphagov/government-frontend/blob/c3b86d9de7be84c86ef8ba86ea759450e604a01d/app/presenters/worldwide_organisation_presenter.rb#L157). Could you possibly link me to the schema change in your publishing API PR that determines this difference?

Comment (brucebolt) [reply app/presenters/worldwide_organisation_presenter.rb:155]: It's possible to have an office without a contact, whereas all the people will have a role (as we're only including those people in the content item).

Comment (hannako) [reply app/presenters/worldwide_organisation_presenter.rb:155]: Ah nice that makes sense, thanks

**Review by hannako — APPROVED** [review]
Looks good! Thanks Bruce!

**Pull request merged**